modules/framework/galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/routes/RunQueryRoute.java

```diff
@@ -279,8 +277,7 @@ private List<IRasSearchCriteria> getCriteria(RasQueryParameters queryParams) thr
 			critList.add(submissionIdCriteria);
 		}
 		if (detail != null && !detail.isEmpty()) {
-			RasSearchCriteriaDetail detailCriteria = new RasSearchCriteriaDetail(detail);
-			critList.add(detailCriteria);
+			rasDetailsQueryParams.setDetail(detail);
```

Comment:
I think this should be done outside this `getCriteria` method since the rest of the method is adding to a list of search criteria, whereas this is setting a detail